src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/cache/CachedLRUObjects.java

```diff
@@ -0,0 +1,44 @@
+package com.cloudbees.jenkins.plugins.bitbucket.server.client.cache;
+
+import java.util.Iterator;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.logging.Logger;
+
+public class CachedLRUObjects<K,V> {
```

Comment:
Why implementing a custom cache solution and not using existing ones like Guava's (https://github.com/google/guava/wiki/CachesExplained)?

Reply:
https://github.com/jenkinsci/bitbucket-branch-source-plugin/pull/91#issuecomment-352396170

As @stephenc said, say no to Guava :sweat: